pkg/broker/filter/filter_handler.go

```diff
@@ -206,34 +219,51 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 
 	h.reportArrivalTime(event, reportArgs)
 
-	h.send(ctx, writer, request.Header, subscriberURI.String(), reportArgs, event, ttl)
+	h.send(ctx, writer, request.Header, subscriberURI.URL(), reportArgs, event, ttl)
 }
 
-func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target string, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
+func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target *url.URL, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
 	// send the event to trigger's subscriber
-	response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
+	dispatchInfo, response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
+
 	if err != nil {
 		h.logger.Error("failed to send event", zap.Error(err))
-		writer.WriteHeader(http.StatusInternalServerError)
-		_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+
+		writer.WriteHeader(dispatchInfo.ResponseCode)
+
+		errExtensionInfo := broker.ErrExtensionInfo{
+			ErrDestination:  target,
+			ErrResponseBody: dispatchInfo.ResponseBody,
+		}
+		errExtensionBytes, msErr := json.Marshal(errExtensionInfo)
+		if msErr != nil {
+			h.logger.Error("failed to marshal errExtensionInfo", zap.Error(err))
+		}
+		_, _ = writer.Write(errExtensionBytes)
```

Comment:
Does this `writer.Write` work when there is a marshal error? Can we handle the error better?

Reply:
oh, yes I need to deal with it